lib/src/exceptions/dart_exception_processor.dart

```diff
@@ -0,0 +1,293 @@
+      stackTrace,
+      inAppIncludes: inAppIncludes,
+      inAppExcludes: inAppExcludes,
+      inAppByDefault: inAppByDefault,
+    );
+
+    final exceptionData = <String, dynamic>{
+      'type': _getExceptionType(error),
+      'mechanism': {
+        'handled': handled,
+        'synthetic': _isPrimitive(error), // we consider primitives as synthetic
+        'type': 'generic',
+      },
+      'thread_id': _getCurrentThreadId(),
+    };
+
+    // Add exception message, if available
+    final errorMessage = error.toString();
+    if (errorMessage.isNotEmpty) {
+      exceptionData['value'] = errorMessage;
+    }
+
+    // Add module from first stack frame (where exception was thrown)
+    final exceptionModule = _getExceptionModule(stackTrace);
+    if (exceptionModule != null && exceptionModule.isNotEmpty) {
+      exceptionData['module'] = exceptionModule;
+    }
+
+    // Add package from first stack frame (where exception was thrown)
+    final exceptionPackage = _getExceptionPackage(stackTrace);
+    if (exceptionPackage != null && exceptionPackage.isNotEmpty) {
+      exceptionData['package'] = exceptionPackage;
+    }
+
+    // Add stacktrace, if any frames are available
+    if (frames.isNotEmpty) {
+      exceptionData['stacktrace'] = {
+        'frames': frames,
+        'type': 'raw',
+      };
+    }
+
+    final result = <String, dynamic>{
+      '\$exception_level': handled ? 'error' : 'fatal',
+      '\$exception_list': [exceptionData],
+    };
+
+    // Add custom properties if provided
+    if (properties != null) {
+      for (final entry in properties.entries) {
+        // Don't allow overwriting system properties
+        if (!result.containsKey(entry.key)) {
+          result[entry.key] = entry.value;
+        }
+      }
+    }
+
+    return result;
+  }
+
+  /// Parses stack trace into PostHog format
+  static List<Map<String, dynamic>> _parseStackTrace(
+    StackTrace stackTrace, {
+    List<String>? inAppIncludes,
+    List<String>? inAppExcludes,
+    bool inAppByDefault = true,
+  }) {
+    final chain = Chain.forTrace(stackTrace);
+    final frames = <Map<String, dynamic>>[];
+
+    for (final trace in chain.traces) {
+      for (final frame in trace.frames) {
+        final processedFrame = _convertFrameToPostHog(
+          frame,
+          inAppIncludes: inAppIncludes,
+          inAppExcludes: inAppExcludes,
+          inAppByDefault: inAppByDefault,
+        );
+        if (processedFrame != null) {
+          frames.add(processedFrame);
+        }
+      }
+    }
+
+    return frames;
+  }
+
+  /// Converts a Frame from stack_trace package to PostHog format
+  static Map<String, dynamic>? _convertFrameToPostHog(
+    Frame frame, {
+    List<String>? inAppIncludes,
+    List<String>? inAppExcludes,
+    bool inAppByDefault = true,
+  }) {
+    final member = frame.member;
+    final fileName = _extractFileName(frame);
+
+    final frameData = <String, dynamic>{
+      'function': member ?? 'unknown',
```

Comment:
i'd rather make it optional in the frontend and backend instead of fallback sdk side